File: LeetCode/ConstructAlgorithm/Golang/Leetcode385-迷你语法分析器.go
package sbgo

import "strconv"

func deserialize(s string) *NestedInteger {
	var dfs func(string) NestedInteger
	u := 0
	dfs = func(s string) NestedInteger {
		res := NestedInteger{}
		if s[u] == '[' {
			u++
			for s[u] != ']' {
				res.Add(dfs(s))
			}
			u++
			if u < len(s) && s[u] == ',' {
				u++
			}
		} else {
			k := u
			for k < len(s) && s[k] != ',' && s[k] != ']' {
				k++
			}
			t, _ := strconv.Atoi(s[u:k])
			res.SetInteger(t)
			if k < len(s) && s[k] == ',' {
				k++
			}
			u = k
		}
		return res
	}
	res := dfs(s)
	return &res
}

File: LeetCode/ConstructAlgorithm/Js/Leetcode385-迷你语法分析器.js
var deserialize = function (s) {
    let u = 0;
    let dfs = function (s) {
        let res = new NestedInteger();
        if (s[u] == '[') {
            u++;
            while (s[u] != ']') res.add(dfs(s))
            u++;
            if (u < s.length && s[u] == ',') u++;
        } else {
            let k = u;
            while (k < s.length && s[k] != ']' && s[k] != ',') k++;
            res.setInteger(s.substr(u, k - u));
            if (k < s.length && s[k] == ',') k++;
            u = k;
        }
        return res;
    }
    return dfs(s)
};